src/utilities/time.ts

```diff
@@ -47,23 +206,167 @@ function parseDurationString(durationString: string): ParsedDurationString | nev
     };
   }
 
-  const fullMicrosecondsRegex = new RegExp(`^${validDurationValueRegex}$`);
+  const isEpoch = validateTime(durationString, TimeTypes.EPOCH);
+  matches = isEpoch ? EPOCH_TIME.exec(durationString) : RELATIVE_TIME.exec(durationString);
+  if (matches !== null) {
+    const [, sign = '', days = '0', hours = '0', minutes = '0', seconds = '0', milliseconds = '0'] =
+      matches as RegExpExecArray;
+
+    const hoursNum = Number(hours);
+    const minuteNum = Number(minutes);
+    const secondsNum = Number(seconds);
+    const millisecondNum = Number(milliseconds);
 
-  if (fullMicrosecondsRegex.test(durationString)) {
-    const microseconds = parseFloat(durationString);
     return {
-      days: 0,
-      hours: 0,
-      isNegative: microseconds < 0,
-      microseconds,
-      milliseconds: 0,
-      minutes: 0,
-      seconds: 0,
+      days: days !== undefined ? Number(days.replace('T', '')) : 0,
+      hours: hoursNum,
+      isNegative: sign !== '' && sign !== '+',
+      microseconds: 0,
+      milliseconds: millisecondNum,
+      minutes: minuteNum,
+      seconds: secondsNum,
       years: 0,
     };
   }
 
-  throw Error('Must be of format: 1y 3d 2h 24m 35s 18ms 70us');
+  // const isEpochSimple = validateTime(durationString, TimeTypes.EPOCH_SIMPLE);
+  matches = new RegExp(`^${validDurationValueRegex}$`).exec(durationString); //isEpochSimple ? EPOCH_SIMPLE.exec(durationString) : RELATIVE_SIMPLE.exec(durationString);
+  if (matches !== null) {
+    const [, sign, num, decimal] = matches as RegExpExecArray;
+    let microsecondNum = 0;
+    let millisecondNum = 0;
+    let secondsNum = 0;
+    let minuteNum = 0;
+    let hourNum = 0;
+    let dayNum = 0;
+    let yearNum = 0;
+
+    //shift everthing based on units
+    switch (units) {
+      case 'microseconds':
+        microsecondNum = Number(num);
```

Comment:
Do you mind using `parseInt` instead of `Number` here to ensure that we only end up with valid integers. Also we can probably pull the parsing of `num` outside before the switch since it's being performed for every case.

Reply:
Sure, I made the changes